protocol/lib/slices.go

```diff
@@ -16,17 +19,17 @@ func ContainsDuplicates[V comparable](values []V) bool {
 	return false
 }
 
-// ConvertMapToSliceOfKeys converts a map to a slice of keys.
-// Note that this method returns the keys in a non-deterministic order.
-func ConvertMapToSliceOfKeys[K comparable, V any](m map[K]V) []K {
-	var slice = make([]K, len(m))
-	index := 0
-	for key := range m {
-		slice[index] = key
-		index += 1
+// GetSortedKeys returns the keys of the map in sorted order.
+func GetSortedKeys[R interface {
```

Comment:
Why do you want an explicit sort interface here since?

You could use [slices.SortFunc()](https://pkg.go.dev/golang.org/x/exp/slices#SortFunc) with a simple func that uses `<` since `k` is comparable?

Reply:
Confusing terminology but [comparable](https://go.dev/blog/comparable) means that you can use `==` to compare the type. I believe what you're thinking of is [constraints.Ordered](https://pkg.go.dev/golang.org/x/exp/constraints#Ordered) which means that `<` can be used.

Reply:
Since this function should be usable for arbitrary structs used as keys, then those types are `comparable` but not `Ordered`